test/unit/pool/basic-tests.js

```diff
@@ -210,6 +211,66 @@ test('pool#connect client.poolCount', function() {
         assert.equal(client.poolCount, undefined,
           'after pool is destroyed, count should be undefined');
       });
-    })
+    });
+  });
+});
+
+pg.defaults.block = false;
+pg.defaults.poolSize = 1;
+
+test('pool#connect acquire errors if the pool is full', function() {
```

Comment:
One of these tests introduces an unnecessarily long timeout, maybe 2 seconds. I'm not super familiar with how the test runner knows when tests are done or I would address it, was hoping you'd know :(